src/Autocomplete/src/Checksum/ChecksumCalculator.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Autocomplete\Checksum;
+
+final class ChecksumCalculator implements ChecksumCalculatorInterface
+{
+    public function __construct(private readonly string $secret)
+    {
+    }
+
+    public function calculateForArray(array $data): string
+    {
+        $this->sortKeysRecursively($data);
+
+        return base64_encode(hash_hmac('sha256', json_encode($data), $this->secret, true));
```

Comment:
Maybe use some lighter hash algo? 
```suggestion
        return base64_encode(hash_hmac('xxh128', json_encode($data), $this->secret, true));
```
Ref: https://php.watch/articles/php-hash-benchmark

Reply:
Would the base64 still be necessary ?

Reply:
Right, I guess the `base64_encode` is kind of a leftover before I started using `hash_hmac` (or I used it to get a shorter string) 🤔. We can remove it and increase the string length from 44 to 64. 

In terms of the algorithm:
![CleanShot 2023-12-25 at 13 42 35@2x](https://github.com/symfony/ux/assets/80641364/7df5f672-e562-4857-8b46-3fdcdb001828)

Reply:
The question here would be how we treat that data, for me it's checksum which is not a password but should be kinda of cryptographic, but it doesn't need to be perfect as a password and it doesn't need to be as safe as it is.

Reply:
Ah, you're right. I thought (idk why :D) that we encrypt data here, but the data is not encrypted at all. 

So the answer here is simple, I used `sha256`, because that's exactly the same way as we calculate the checksum in the LiveComponent. I'm open to changing it, I just copied the code from the LiveComponent and didn't bother to think whether `sha256` is the correct one here :D.

Reply:
Now I'm thinking about extracting checksum calculating inside the LiveComponent component, and using it in the autocomplete component. Autocomplete depends on the LiveComponent so it might make sense.

Reply:
That would make perfect sense to me. 

Feel free to do it in a second/later PR if you want to release this one first... in all maner this should be "internal" so there is no BC involved there :)

Reply:
I was 100% sure `LiveComponent` is required by the `Autocomplete`, but it is not.

So, for now, I leave the `Calculator` where it lives. I can mark the class as `internal` but I'll leave the interface as it is, to allow people to create their own implementation of checksum calculator. If anyone finds they need a lighter version of hash algorithm, it'll make sense to create their own.

Reply:
> but I'll leave the interface as it is,

I'd prefer to remove the interface. I can't see a legit reason to need to write your own. If there is a problem with our class, then we want them to tell us so we can fix it :)

Reply:
@weaverryan removed!